File: leetcode/100-999/125. Valid Palindrome/solution_1.js
/**
 * @param {string} s
 * @return {boolean}
 */
 
var toLowerChars = function(s)
{
    let chars = ""

    for (const ch of s.toLowerCase())
    {
        if (/[a-z0-9]/.test(ch))
            chars += ch;
    }

    return chars;
}

/**
 * @param {string} s
 * @return {boolean}
 */

var isPalindrome = function(s) 
{
    const chars = toLowerChars(s)
    const reversedChars = chars.split("").reverse().join("");

    return chars == reversedChars
};

File: leetcode/100-999/125. Valid Palindrome/solution_1.cpp
class Solution 
{
    public:
        string toLowerLetters(string s)
        {
            string chars;

            for (auto ch : s)
            {
                if (isalpha(ch) || isdigit(ch))
                    letters += tolower(ch);
            }

            return chars;
        }

        bool isPalindrome(string s) 
        {
            string letters = toLowerLetters(s);
            string reversedLetters = toLowerLetters(s);
            
            reverse(reversedLetters.begin(), reversedLetters.end());

            return reversedLetters == letters;
        }
};